### ISSUE TITLE
confuse about result

### ISSUE THREAD

**Issue by Johnson-yue** [POSTECH-CVLab/PyTorch-StudioGAN#1]
Hi， thank you for your sharing the good repo for reimplement for GAN.

I run the command :
`python main.py --eval -t -c "./configs/Table2/biggan32_cifar_hinge_no.json"`
the result is 
> model=G-step=80000-Inception_mean=9.465-Inception_std=0.00551-FID=10.344.pth

but when I use contraGAN with the same dataset

`python main.py --eval -t -c "./configs/Table2/contragan32_cifar_hinge_no.json"`

the result is:
> Current best model (FID) is  .....  step=78000-Inception_mean=8.89-Inception_std=0.1667-FID=12.648

when  contraGAN is not better than biggan?

**Comment by MINGUKKANG** [comment]
Thank you for your interest. 
Could you show me the log file of the below contraGAN experiment (or send mgkang@postech.ac.kr it)? 
It may be located in ./logs/{run_name}.log.
Also, I rerun Table2/contragan32_cifar_hinge_no.json experiment now, I will tell you as soon as I find the problem.

I think that there seems to be something wrong with the process of code refactoring:)

**Comment by Johnson-yue** [comment]
@MINGUKKANG , ok, I check it for later, and I will send log to you

**Comment by Johnson-yue** [comment]
I find the log file and have sent to you

**Comment by MINGUKKANG** [comment]
Can I ask you a PyTorch Version and OS(Window or Ubuntu) that you were used to experiment with ContraGAN? 
Since Training GAN is extremely unstable, I think that matching the PyTorch version and OS can play a very important role in reproducing the performance of GANs. 

In my case, 

PyTorch 1.3.1
OS Ubuntu 18.04
TorchVision 0.4.2
Multi-GPU False (Only CIFAR10)

Also, I am doing the training ContraGAN in my environment and can identify the result the next day.
I will post it ASAP.

Sincerely, mgkang

**Comment by Johnson-yue** [comment]
PyTorch 1.5.0
Python 3.7.2
OS Ubuntu 18.04
TorchVision 0.6.0
Multi-GPU False (Only CIFAR10)

**Comment by MINGUKKANG** [comment]
Hi @Johnson-yue 

I retrained ContraGAN using "./configs/Table2/contragan32_cifar_hinge_no.json" configuration, and the ContraGAN gives 10.287 FID on CIFAR10 dataset. You can see the log file below. 

I think you were just not lucky. May I ask you to train ContraGAN using another seed? 
Also, I recommend that you experiment in the same environment as when you train cGAN.
When I read the log generated from your ContraGAN training, training time was doubled to cGAN training time.
I don't know if you trained this way, but I've noticed a slight performance degradation when training two models in only one GPU.

[Log file]
[contragan32_cifar_hinge_no-train-2020_06_30_17_02_29.log](https://github.com/POSTECH-CVLab/PyTorch-StudioGAN/files/4856036/contragan32_cifar_hinge_no-train-2020_06_30_17_02_29.log)

I hope my answer would help you to solve the problem.
Sincerely, Minguk Kang

**Comment by MINGUKKANG** [comment]
If you can not solve the problem after doing so, please don’t hesitate to leave a comment:)

**Comment by Johnson-yue** [comment]
When I train cGAN, my GPU only has this model in training, but when I train contraGAN, my GPU also has another model training at the same time, so the time is relatively long. Maybe this is the main reason for the low score ？

I will retrain contraGAN again, and if I have solved this problem , I will close this issue

and thank you for your attention

**Comment by MINGUKKANG** [comment]
Hi

Although I do not know why this phenomenon occurs, I have had a similar experience with you. I will cross my fingers for you.

If you can not solve the problem after doing so, please don’t hesitate to leave a comment!

**Comment by Johnson-yue** [comment]
I rerun contragan32， and got the result is similar to you， but lower 。 I think it is enough to verification this method is work， thank you

I got best FID = 10.398， best IS=9.516

here is my log：
[contragan32_cifar_hinge_no-train-2020_07_03_16_22_30.log](https://github.com/POSTECH-CVLab/PyTorch-StudioGAN/files/4876462/contragan32_cifar_hinge_no-train-2020_07_03_16_22_30.log)

thanks， I will test in my network and my own dataset， if I have some  result ， I will call you

**Comment by Johnson-yue** [comment]
@MINGUKKANG  Hi， I add Diff-Augmentation trick to train contraGAN32， and got：
best FID = 9.2776 
best IS = 9.585

here is log， if you need I can upload the code
[contragan32_cifar_diffaug_hinge_no-train-2020_07_07_18_09_28.log](https://github.com/POSTECH-CVLab/PyTorch-StudioGAN/files/4888815/contragan32_cifar_diffaug_hinge_no-train-2020_07_07_18_09_28.log)

**Comment by MINGUKKANG** [comment]
Hi:)

Thank you so much for your interest in our ContraGAN and for conducting additional experiments:)

Can I ask you to commit your code for the improvement of our work?

Sincerely, Minguk Kang

**Comment by Johnson-yue** [comment]
ok, I will commit later

**Comment by Johnson-yue** [comment]
I test the “consistency regularization”  trick，but maybe I am not lucky。
I got best` IS=9.316， best FID = 10.566`

log is here
[contragan32_cifar_diffaug_cr_hinge_no-train-2020_07_13_18_46_25.log](https://github.com/POSTECH-CVLab/PyTorch-StudioGAN/files/4917092/contragan32_cifar_diffaug_cr_hinge_no-train-2020_07_13_18_46_25.log)
